### PR TITLE
Refactored random number getting code, added random from array and object

### PR DESCRIPTION
**Pull request by Miodec**

### CODE REVIEW

File: frontend/src/scripts/utils/misc.ts [hunk 1]
```diff
@@ -997,3 +997,13 @@ export function shuffle<T>(elements: T[]): void {
     elements[i] = temp;
   }
 }
+
+export function randomElementFromArray<T>(array: T[]): T {
+  return array[randomIntFromRange(0, array.length - 1)];
```
Comment (Bruception) [inline frontend/src/scripts/utils/misc.ts:1002]: How does this behave if array is empty?

Comment (Miodec) [reply frontend/src/scripts/utils/misc.ts:1002]: Currently it will return `undefined`

File: frontend/src/scripts/utils/misc.ts [hunk 2]
```diff
@@ -997,3 +997,13 @@ export function shuffle<T>(elements: T[]): void {
...
+}
+
+export function randomElementFromObject(object: {
```
Comment (Bruception) [inline frontend/src/scripts/utils/misc.ts:1005]:
```ts
randomElementFromObject<T extends object>(obj: T): T[keyof T]
```

Comment (Miodec) [reply frontend/src/scripts/utils/misc.ts]: Done

File: frontend/src/scripts/utils/misc.ts (hunk 1 above)
Comment (Bruception) [reply frontend/src/scripts/utils/misc.ts:1002]: Should be fine as long as we handle this possibility when this function is used.

**Review by Bruception — APPROVED** [review]
LGTM IIW

**Review by Ferotiq — APPROVED** [review]
LGTM also

**Pull request merged**